diff --git a/lib/sqlalchemy/dialects/mysql/base.py b/lib/sqlalchemy/dialects/mysql/base.py
index ea180eee6..cea62e93d 100644
--- a/lib/sqlalchemy/dialects/mysql/base.py
+++ b/lib/sqlalchemy/dialects/mysql/base.py
@@ -2716,9 +2716,19 @@ class _DecodingRowProxy(object):
     # sets.Set(['value']) (seriously) but thankfully that doesn't
     # seem to come up in DDL queries.
 
+    _encoding_compat = {
+        "utf16": "utf-16-be",
+        "utf8mb4": "utf8",
+        "utf8mb3": "utf8",
+    }
+
     def __init__(self, rowproxy, charset):
         self.rowproxy = rowproxy
-        self.charset = charset
+        self.charset = (
+            self._encoding_compat.get(charset, charset)
+            if charset is not None
+            else None
+        )
 
     def __getitem__(self, index):
         item = self.rowproxy[index]
